From f9fac7f7770f056790b628e47af91ed3e814246c Mon Sep 17 00:00:00 2001
From: antirez <antirez@gmail.com>
Date: Wed, 5 Jul 2017 15:40:25 +0200
Subject: [PATCH] Avoid closing invalid fds

These fds can be invalid if, for example, we fail to open ns/mnt.

diff --git a/VolumeManager.cpp b/VolumeManager.cpp
index c3c95f6..6098f9a 100644
--- a/VolumeManager.cpp
+++ b/VolumeManager.cpp
@@ -688,8 +688,12 @@
         }
 
 next:
-        close(nsFd);
-        close(pidFd);
+        if (nsFd >= 0) {
+            close(nsFd);
+        }
+        if (pidFd >= 0) {
+            close(pidFd);
+        }
     }
     closedir(dir);
     return 0;
